addOns/openapi/src/main/java/org/zaproxy/zap/extension/openapi/ExtensionOpenApi.java

```diff
@@ -343,9 +344,13 @@ public OpenApiResults importOpenApiDefinitionV2(
                 throw new InvalidDefinitionException();
             }
 
+            String openApiString = FileUtils.readFileToString(file, "UTF-8");
+
             List<String> errors =
                     importOpenApiDefinition(
-                            Json.pretty(openApi),
+                            !openApiString.contains("openapi:") || openApiString.contains(".yaml#")
```

Comment:
This checks will not work for JSON format.

Reply:
Make the colon optional then it should still work shouldn't it?

Reply:
As regex

Reply:
But we are just workaround the issue (and maybe), better to find out how to (potentially) fixing it earlier. Though we can merge as is as it helps in some cases and investigate more later.

Reply:
I'm not sure how to fix earlier as is is failing in Json.pretty which is outside of ZAP